File: go/cars-assemble/cars_assemble.go
package cars

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	x := float64(productionRate)
	y := successRate * 0.01
	return x * y
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	x := CalculateWorkingCarsPerHour(productionRate, successRate)
	return int(x / 60)
}

// CalculateCost works out the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	groups := float64(carsCount / 10)
	remaining := float64(carsCount % 10)
	return uint(groups * 95000 + remaining * 10000)
}

File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (cond (>= speed 10) (* (* 221.0 speed) 0.77)
        (= speed 9) (* (* 221.0 speed) 0.8)
        (>= speed 5) (* (* 221.0 speed) 0.9)
        (>= speed 1) (* 221.0 speed)
        :else 0.0
        )
  )

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) 60))
  )
